tests/beaker_tests/netdev_stdlib/network_dns_provider_nondefaults.rb

```diff
@@ -0,0 +1,252 @@
+  ## @step [Step] Sets up switch for provider test.
+  step 'TestStep :: Setup switch for provider test' do
+    # Define PUPPETMASTER_MANIFESTPATH constant using puppet config cmd.
+    UtilityLib.set_manifest_path(master, self)
+
+    # Expected exit_code is 0 since this is a vegas shell cmd.
+    cmd_str = UtilityLib.get_vshell_cmd('conf t ; ' \
+                                        'no ip domain-name switch1.test.com ; ' \
+                                        'no ip domain-name switch2.test.com ; ' \
+                                        'no ip domain-list test.com ; ' \
+                                        'no ip domain-list test.net ; ' \
+                                        'no ip name-server 8.8.8.8 ; ' \
+                                        'no ip name-server 2001:4860:4860::8888')
+    on(agent, cmd_str)
+
+    # Expected exit_code is 0 since this is a vegas shell cmd.
+    # Flag is set to true to check for absence of RegExp pattern in stdout.
+    cmd_str = UtilityLib.get_vshell_cmd('show running-config')
+    on(agent, cmd_str) do
+      UtilityLib.search_pattern_in_output(stdout, [/name-server .*8\.8\.8\.8$/],
+                                          true, self, logger)
+    end
+    logger.info("Setup switch for provider test :: #{result}")
+  end
+
+  # @step [Step] Requests manifest from the master server to the agent.
+  step 'TestStep :: Set the properties in a manifest from master' do
+    # Expected exit_code is 0 since this is a bash shell cmd.
+    on(master, NetworkDnsLib.create_network_dns_manifest('switch1.test.com',
+                                                         ['test.com', 'test.net'],
+                                                         ['8.8.8.8', '2001:4860:4860::8888'],
+                                                        ))
+
+    # Expected exit_code is 2 since this is a puppet agent cmd with change.
+    cmd_str = UtilityLib.get_namespace_cmd(agent, UtilityLib::PUPPET_BINPATH +
+      'agent -t', options)
+    on(agent, cmd_str, acceptable_exit_codes: [2])
+
+    logger.info("Set the domain property in a manifest from master :: #{result}")
+  end
+
+  # @step [Step] Checks network_dns resource on agent using resource cmd.
+  step 'TestStep :: Check network_dns resource on agent' do
+    # Expected exit_code is 0 since this is a puppet resource cmd.
+    # Flag is set to false to check for presence of RegExp pattern in stdout.
+    cmd_str = UtilityLib.get_namespace_cmd(agent, UtilityLib::PUPPET_BINPATH +
+      "resource network_dns 'settings'", options)
+    on(agent, cmd_str) do
+      UtilityLib.search_pattern_in_output(stdout,
+                                          { 'ensure'  => 'present',
+                                            'domain'  => 'switch1.test.com',
+                                            'search'  => "\\['test.com', 'test.net'\\]",
+                                            'servers' => "\\['8.8.8.8', '2001:4860:4860::8888'\\]" },
```

Comment:
Hmm, OK. Probably something that needs to be changed in our library code - I'll take a note to look into this as time permits, but fine for now.